python_files/ml_on_iris_dataset.py

```diff
@@ -0,0 +1,192 @@
+# Iris Machine Learning Assignment 1
+# Will McGrath
+# September 8, 2020
+
+import sys
+
+import numpy as np
+import pandas as pd
+from plotly import express as px
+from plotly import graph_objects as go
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import log_loss, roc_auc_score
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import Normalizer, OneHotEncoder
+from sklearn.svm import LinearSVC
+
+
+def print_heading(title):
+    # creates headers to divide outputs
+    print("\n")
+    print("*" * 80)
+    print(title)
+    print("*" * 80)
+
+    return None
+
+
+def main():
+    # pulls in the Iris dataset from a CSV and creates a dataframe
+    print_heading("Pulling in Data and Creating Dataframe")
+    data = pd.read_csv("./data/iris.data", header=None)
+    data_df = pd.DataFrame(data).rename(
+        columns={
+            0: "sepal_len_cm",
+            1: "sepal_wid_cm",
+            2: "petal_len_cm",
+            3: "petal_wid_cm",
+            4: "class",
+        }
+    )
+    print(data_df)
+
+    # summary statistics with numpy example
+    print_heading("Dataframe Summary Statistics")
+    print(data_df.describe())
+    assert (
```

Comment:
I'm confused by this line. Is this asserting mean values in the describe table? Doesn't describe already show mean?

Reply:
Yes, describe already shows mean. I just added the assert method to show that getting the mean using numpy and getting the mean from describe() is the same thing. I should make that more clear in my comments.